### ISSUE TITLE
Degrade with Ubuntu 22.04 while printing

### ISSUE THREAD

**Issue by chennasudhakar94** [OpenPrinting/cups-filters#473]

**Comment by zdohnal** [comment]
@tillkamppeter would you mind looking into it? Is this an expected behavior of pdftopdf?

**Comment by tillkamppeter** [comment]
In the logs I see that the job has not received a "media" or "PageSize" option:
```
D [21/Jun/2022:15:44:50 +0530] [Job 18] argv[5]="finishings=3 number-up=1 job-uuid=urn:uuid:c8fdd2fd-be68-3bd8-45c0-14ee14613bd7 job-originating-host-name=localhost date-time-at-creation= date-time-at-processing= time-at-creation=1655806490 time-at-processing=1655806490 document-name-supplied=3Pages_A3_Landscape.pdf"
```
Usually there is such option, even if you do not explicitly supply it.

Either something has changed in the client application or in CUPS (@zdohnal @michaelrsweet).

**Comment by tillkamppeter** [comment]
Also note,that `pdftopdf` never had the functionality of overtaking the size from each input page, instead, it was always taking the default size if no size was supplied. In cups-filters 2.x this will change, so that the page sizes of the input file actually get overtaken.

**Comment by zdohnal** [comment]
@tillkamppeter no such option was applied in cups 2.3.1, where this 'autodetected page size printing' worked - see the log from [this comment](https://github.com/OpenPrinting/cups-filters/issues/473#issuecomment-1167056447).

If it is not defined, I would expect PPD default is taken (which happens in 2.4.1). The only change I was able to find was the additional log from pdftopdf for scaling, so my guess was pdftopdf scales the input to PPD default.

**Comment by tillkamppeter** [comment]
Yes, `pdftopdf` scales down pages which are too large for the output page size (the default page size if no other size is explicitly selected). So to print on A3 paper (if it is loaded in the printer) the destination page size needs to get explicitly selected as A3, via command line or via print dialog, or one changes the default pager size to A3 (for example if one uses only A3 on the printer).

**Comment by chennasudhakar94** [comment]
Hi @tillkamppeter,
Following are cups on two OS.
Ubuntu 22.04 is using cups version = 2.4.1
Ubuntu 20.04 is using cups version = 2.3.1

And I see different behaviour on two OS.

Thanks & Regards,
C.Sudhakar

**Comment by chennasudhakar94** [comment]
Hi @tillkamppeter/@zdohnal,

 Can I know what could be the reason is it's cups version?

Thanks & Regards,
C.Sudhakar

**Comment by chennasudhakar94** [comment]
Hi @tillkamppeter/@zdohnal,

Any update on this issue.

Thanks & Regards,
C.Sudhakar

**Comment by chennasudhakar94** [comment]
Hi @tillkamppeter/@zdohnal,

Any update on this issue.

Thanks & Regards,
C.Sudhakar
